x/ibc/07-tendermint/misbehaviour_test.go

```diff
@@ -31,58 +32,58 @@ func (suite *TendermintTestSuite) TestCheckMisbehaviour() {
+		consensusState ibctmtypes.ConsensusState
+		evidence       ibctmtypes.Evidence
 		height         uint64
 		expPass        bool
 	}{
 		{
 			"valid misbehavior evidence",
-			tendermint.ClientState{ID: chainID, LatestHeight: height, FrozenHeight: 0},
-			tendermint.ConsensusState{Root: commitment.NewRoot(tmhash.Sum([]byte("app_hash"))), ValidatorSet: bothValSet},
-			tendermint.Evidence{
-				Header1:  tendermint.CreateTestHeader(chainID, height, suite.now, bothValset, suite.valSet, bothSigners),
-				Header2:  tendermint.CreateTestHeader(chainID, height, suite.now, bothValset, bothValSet, bothSigners),
+			ibctmtypes.NewClientState(chainID, trustingPeriod, ubdPeriod, height, suite.now),
+			ibctmtypes.ConsensusState{Timestamp: suite.now, Root: commitment.NewRoot(tmhash.Sum([]byte("app_hash"))), ValidatorSet: bothValSet},
+			ibctmtypes.Evidence{
+				Header1:  ibctmtypes.CreateTestHeader(chainID, height, suite.now, bothValSet, suite.valSet, bothSigners),
+				Header2:  ibctmtypes.CreateTestHeader(chainID, height, suite.now, bothValSet, bothValSet, bothSigners),
 				ChainID:  chainID,
 				ClientID: chainID,
 			},
 			height,
 			true,
 		},
 		{
-			"height doesn't match provided evidence",
-			tendermint.ClientState{ID: chainID, LatestHeight: height, FrozenHeight: 0},
-			tendermint.ConsensusState{Root: commitment.NewRoot(tmhash.Sum([]byte("app_hash"))), ValidatorSet: bothValSet},
-			tendermint.Evidence{
-				Header1:  tendermint.CreateTestHeader(chainID, height, suite.now, bothValset, suite.valSet, bothSigners),
-				Header2:  tendermint.CreateTestHeader(chainID, height, suite.now, bothValset, bothValSet, bothSigners),
+			"valid misbehavior at height greater than last consensusState",
+			ibctmtypes.NewClientState(chainID, trustingPeriod, ubdPeriod, height-1, suite.now),
+			ibctmtypes.ConsensusState{Timestamp: suite.now, Height: height - 1, Root: commitment.NewRoot(tmhash.Sum([]byte("app_hash"))), ValidatorSet: bothValSet},
+			ibctmtypes.Evidence{
+				Header1:  ibctmtypes.CreateTestHeader(chainID, height, suite.now, bothValSet, suite.valSet, bothSigners),
+				Header2:  ibctmtypes.CreateTestHeader(chainID, height, suite.now, bothValSet, bothValSet, bothSigners),
 				ChainID:  chainID,
 				ClientID: chainID,
 			},
-			0,
-			false,
-		},
-		{
-			"consensus state's valset hash different from evidence",
-			tendermint.ClientState{ID: chainID, LatestHeight: height, FrozenHeight: 0},
-			tendermint.ConsensusState{Root: commitment.NewRoot(tmhash.Sum([]byte("app_hash"))), ValidatorSet: suite.valSet},
-			tendermint.Evidence{
-				Header1:  tendermint.CreateTestHeader(chainID, height, suite.now, bothValset, suite.valSet, bothSigners),
-				Header2:  tendermint.CreateTestHeader(chainID, height, suite.now, bothValset, bothValSet, bothSigners),
-				ChainID:  chainID,
-				ClientID: chainID,
-			},
-			height,
-			false,
+			height - 1,
+			true,
 		},
+		// {
+		// 	"consensus state's valset hash different from evidence",
```

Comment:
uncomment ?

Reply:
This test case should actually pass without error now, uncommenting